colossalai/inference/modeling/models/nopadding_llama.py

```diff
@@ -335,44 +339,47 @@ def _load_from_state_dict(
     ):
         # NOTE This is a hack to ensure we could load the right weight from LlamaMLP checkpoint due to the use of torch.stack(gate_weight, up_weight)
 
-        for hook in self._load_state_dict_pre_hooks.values():
-            hook(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs)
-
-        persistent_buffers = {k: v for k, v in self._buffers.items() if k not in self._non_persistent_buffers_set}
-        local_name_params = itertools.chain(self._parameters.items(), persistent_buffers.items())
-        local_state = {k: v for k, v in local_name_params if v is not None}
+        if hasattr(self, "gate_up_dict"):
```

Comment:
1. With `self.gate_up_dict` already set in `__init__`, the if block is noneffective and redundant. 
2. prevent hard-coding

Reply:
Because of the problems we discussed before, q,k,v will be stored in diff checkpoint files, so hardcode is inevitable :(.